packages/api/src/graphql/comment-rating/comment-rating.private-mutation.ts

```diff
@@ -0,0 +1,70 @@
+import {PrismaClient, RatingSystemType} from '@prisma/client'
+import {Context} from '../../context'
+import {authorise, CanUpdateCommentRatingSystem} from '../permissions'
+
+type UpdateCommentRatingAnswer = {id: string; answer: string; type: RatingSystemType}
+
+export const updateRatingSystem = (
+  ratingSystemId: string,
+  name: string | undefined,
+  answers: UpdateCommentRatingAnswer[] | undefined,
+  authenticate: Context['authenticate'],
+  ratingSystem: PrismaClient['commentRatingSystem']
+) => {
+  const {roles} = authenticate()
+  authorise(CanUpdateCommentRatingSystem, roles)
+
+  return ratingSystem.update({
+    where: {id: ratingSystemId},
+    data: {
+      name,
+      answers: {
+        update: answers?.map(answer => ({
+          where: {id: answer.id},
+          data: {
+            answer: answer.answer
+          }
+        }))
+      }
+    },
+    include: {
+      answers: true
+    }
+  })
+}
+
+export const createCommentRatingAnswer = async (
+  ratingSystemId: string,
+  type: RatingSystemType,
+  answer: string | undefined,
+  authenticate: Context['authenticate'],
+  ratingAnswer: PrismaClient['commentRatingSystemAnswer']
+) => {
+  const {roles} = authenticate()
+  authorise(CanUpdateCommentRatingSystem, roles)
```

Comment:
Should be "CanCreateCommentRatingSystem"

Reply:
CanUpdateCommentRatingSystem is correct as this is updating the system